src/BlockActions.js

```diff
@@ -0,0 +1,181 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import {bindActionCreators} from 'redux';
+import {connect} from 'react-redux';
+import {
+  getLabel,
+  getNestedBlockDefinition,
+  getSiblingsIds, isSimpleLayout,
+  triggerCustomEvent, triggerKeyboardEvent
+} from './processing/utils';
+import {duplicateBlock, hideBlock} from './actions';
+import {refType} from './types';
+
+
+@connect((state, props) => {
+  const {fieldId, blockId} = props;
+  const blockDefinition = getNestedBlockDefinition(state, fieldId, blockId);
+  const siblings = getSiblingsIds(state, fieldId, blockId);
+  return {
+    isSimpleLayout: isSimpleLayout(blockDefinition, state[fieldId].isMobile),
+    blockDefinition,
+    siblings,
+    icons: state[fieldId].icons,
+    index: siblings.indexOf(blockId),
+  };
+}, (dispatch, props) => {
+  const {fieldId, blockId} = props;
+  return bindActionCreators({
+    hideBlock: () => hideBlock(fieldId, blockId),
+    duplicateBlock: () => duplicateBlock(fieldId, blockId),
+  }, dispatch);
+})
```

Comment:
This component could receive its data from `Block`, instead of also be connected to Redux.